core/domain/skill_services.py

```diff
@@ -514,3 +515,107 @@ def save_skill_summary(skill_summary):
     )
 
     skill_summary_model.put()
+
+
+def create_user_skill_mastery(user_id, skill_id, degree_of_mastery):
+    """Creates skill mastery of a user.
+
+    Args:
+        user_id: str. The user id of the logged in user.
+        skill_id: str. The unique id of the skill.
+        degree_of_mastery: float. The degree of mastery of user in the skill.
+    """
+    model_id = user_models.UserSkillMasteryModel.get_model_id(
+        user_id, skill_id)
+
+    user_skill_mastery = skill_domain.UserSkillMastery(
+        model_id, user_id, skill_id, degree_of_mastery)
+    save_user_skill_mastery(user_skill_mastery)
+
+
+def save_user_skill_mastery(user_skill_mastery):
+    """Stores skill mastery of a user.
+
+    Args:
+        user_skill_mastery: dict. The user skill mastery model of a user.
+    """
+    user_skill_mastery_model = user_models.UserSkillMasteryModel(
+        id=user_skill_mastery.id,
+        user_id=user_skill_mastery.user_id,
+        skill_id=user_skill_mastery.skill_id,
+        degree_of_mastery=user_skill_mastery.degree_of_mastery)
+
+    user_skill_mastery_model.put()
+
+
+def get_skill_mastery(user_id, skill_id):
+    """Fetches the mastery of user in a particular skill.
+
+    Args:
+        user_id: str. User id of the user currently logged in.
+        skill_id: str. Unique id of the skill for which mastery degree is
+            requested.
+
+    Returns:
+        degree_of_mastery: float. Mastery degree of the user for the
+            requested skill.
+    """
+    degree_of_mastery = user_models.UserSkillMasteryModel.get(
+        user_models.UserSkillMasteryModel.get_model_id(
+            user_id, skill_id)).degree_of_mastery
+
+    return degree_of_mastery
+
+
+def get_multi_skill_mastery(user_id, skill_ids):
+    """Fetches the mastery of user in multiple skills.
+
+    Args:
+        user_id: str. User id of the user currently logged in.
+        skill_ids: list(str). Skill IDs of the skill for which mastery degree is
+            requested.
+
+    Returns:
+        degree_of_mastery: list(float). Mastery degree of the user for requested
+            skills.
+    """
+    degrees_of_mastery = []
+    model_ids = []
+
+    for skill_id in skill_ids:
+        model_ids.append(
+            user_models.UserSkillMasteryModel.get_model_id(user_id, skill_id))
+
+    skill_mastery_models = user_models.UserSkillMasteryModel.get_multi(
+        model_ids)
+    for skill_summary in skill_mastery_models:
```

Comment:
We've talked about this; this is not a skill summary. It should be skill_mastery_model.

Please proofread PRs carefully before submitting and ensure you have addressed all prior comments, otherwise the review cycle takes longer. Thanks!

Reply:
Sorry about that